### PR TITLE
Roster

### PR DESCRIPTION
**Pull request by mpollmeier**

### CODE REVIEW

File: src/main/scala/ylabs/messaging/Client.scala [hunk 1]
```diff
@@ -89,25 +92,25 @@ class Client extends FSM[State, Context] {
...
+    case Event(msg: Messages.ListenerEvent, ctx) ⇒
+      ctx.eventListeners foreach { _ ! msg }
       stay
 
-    case Event(Messages.SendMessage(recipient, message), ctx) if ctx.chatManager.isDefined ⇒
-      val chat = ctx.chats.get(recipient)
-        .getOrElse(createChat(ctx.chatManager.get, recipient))
+    case Event(Messages.SendMessage(recipient, message), ctx @ Context(Some(connection), chats, _)) ⇒
+      val chat = chats.get(recipient)
+        .getOrElse(createChat(connection, recipient))
```
Comment (KadekM) [inline src/main/scala/ylabs/messaging/Client.scala]: you can use straight getOrElse on map

```
  val chat = chats.getOrElse(key = recipient, createChat(connection, recipient))
```

**Comment by KadekM** [comment]
LGTM

**Comment by mpollmeier** [comment]
@KadekM thank you!

File: src/main/scala/ylabs/messaging/Client.scala (hunk 1 above)
Comment (mpollmeier) [inline src/main/scala/ylabs/messaging/Client.scala]: nice one. done.

**Pull request merged**